src/Posts/RepostButton.jsx

```diff
@@ -0,0 +1,170 @@
+const item = props.item;
+
+if (!item) {
+  return "";
+}
+
+const reposts = Social.index("repost", item);
+
+const dataLoading = reposts === null;
+
+const repostsByUsers = Object.fromEntries(
+  (reposts || []).filter((repost) => repost.value.type === "repost").map((repost) => [repost.accountId, repost]),
+);
+
+if (state.hasRepost === true) {
+  repostsByUsers[context.accountId] = {
+    accountId: context.accountId,
+  };
+}
+
+const accountsWithReposts = Object.keys(repostsByUsers);
+const hasRepost = context.accountId && !!repostsByUsers[context.accountId];
```

Comment:
Curious to know the intention behind a `state.hasRepost` and this locally scoped `hasRepost`

Reply:
see updated commit notes.